### PR TITLE
Enforce formatting

### PR DESCRIPTION
**Pull request by pascutto**
Use dune for formatting, and add it to tests so that the CI fails if the code does not comply (including examples and tests).
We should wait until vendors are removed, because they don't use the same formatting hence the CI will fail.

### CODE REVIEW

**Comment by emillon** [comment]
Formatting parameters in dune are per-project, so it should ignore vendored directories. Does it work?

**Comment by pascutto** [comment]
You can see the test failing there : https://travis-ci.org/pascutto/wodan/jobs/498660713
Says `vendor/mirage-block-solo5/src/block.mli` is not well formatted.

**Comment by g2p** [comment]
Probably because mirage-block-solo5 is not dunified.

**Comment by emillon** [comment]
Dunifying `mirage-block-solo5` will fix this (we can work on that together), but in the meantime you can add a `dune-project` file in `vendor/` with a name like `(other-vendored-projects)` and that should make dune attach all these files to it.

**Comment by g2p** [comment]
+1 for Étienne's solution.
My plan for this would be to merge #53 which removes a few vendored dependencies (the indirect ones we don't need anymore), merge a version of #47 which removes more vendored stuff but doesn't touch the non-Irmin related bits, and finally merge this while ignoring mirage-block / mirage-block-solo5 / mirage-block-unix until they are dunified.

**Comment by pascutto** [comment]
`mirage-block`, `mirage-block-solo5` and `mirage-block-unix` now all use `dune`, so I guess the forks can be updated accordingly.

**Comment by pascutto** [comment]
Rebased and ready for review.
Always using the latest version of ocamlformat was obviously not a good idea in the first place. Besides the breaking change of last week, we don't want formatting to change (or make the CI fail) so often.
I propose that we use https://github.com/ocaml-ppx/ocamlformat/commit/2de7c5f fixed version while `0.9` is not released.

File: examples/solo5-irmin/dispatch.ml [hunk 1]
```diff
@@ -6,56 +6,78 @@ module type HTTP = Cohttp_lwt.S.Server
...
 module BlockCon = struct
-  include Block (* from mirage-block-solo5 *)
-  let discard _ _ _ = Lwt.return @@ Ok ()
-end
+  include Block
 
-module DB = Wodan_irmin.DB_BUILDER
-  (BlockCon)(Wodan.StandardSuperblockParams)
-module Wodan_Git_KV = Wodan_irmin.KV_git(DB)
+  (* from mirage-block-solo5 *)
```
Comment (g2p) [inline examples/solo5-irmin/dispatch.ml:13]: Moving the comment two lines down changes its meaning.
@gpetiot, is it expected that https://github.com/ocaml-ppx/ocamlformat/commit/2de7c5f would move comments?
Do you recommend a different snapshot?

Comment (pascutto) [reply examples/solo5-irmin/dispatch.ml:13]: Good catch, I fixed it here.

Comment (pascutto) [reply examples/solo5-irmin/dispatch.ml:13]: Reported in https://github.com/ocaml-ppx/ocamlformat/issues/660

**Comment by pascutto** [comment]
I'm not sure why the CI is marked failing, it's actually passing.

**Review by g2p — APPROVED** [review]
Looks good.

**Pull request merged**